python/pyspark/sql/functions.py

```diff
@@ -1161,7 +1161,7 @@ def to_date(col, format=None):
 @since(2.2)
 def to_timestamp(col, format=None):
     """Converts a :class:`Column` of :class:`pyspark.sql.types.StringType` or
-    :class:`pyspark.sql.types.TimestampType` into :class:`pyspark.sql.types.DateType`
+    :class:`pyspark.sql.types.DateType` into :class:`pyspark.sql.types.TimestampType`
```

Comment:
Cannot the function accept values of `TimestampType` as well? Internally, the first parameter should be casted to `TimestampType`. If it is already a timestamp, should be ok.

Reply:
Hi @MaxGekk! Yes, it can. The other alternative is:

```python3
def to_timestamp(col, format=None):
    """Converts a :class:`Column` of :class:`pyspark.sql.types.StringType`, :class:`pyspark.sql.types.DateType`, or
    :class:`pyspark.sql.types.TimeStampType` into :class:`pyspark.sql.types.TimestampType`
```

In that case, the documentation for `to_date` should also be updated to explicitly say that it converts ```:class:`pyspark.sql.types.DateType` into :class:`pyspark.sql.types.DateType` ```.

I chose to leave these two cases out because

1. that's how `to_date` does it
2. they're "trivial" cases

Happy to update both as above if you prefer, though.

Reply:
`CAST` can accept not only `DateType` but others, see https://github.com/apache/spark/blob/e0efd213eb6ff61ce2b0f54f55201899fa4171a6/sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/Cast.scala#L423-L447
you can:
1. mention all those types in the comment, or
2. say that any types supported by `CAST` while converting to `TimestampType`, for instance `DateType`.

Reply:
I think easier way might be 2. I would just do as below

```
Converts a :class:`Column` into :class:`pyspark.sql.types.TimestampType`.
It works as a synonym of ``CAST(col AS TIMESTAMP)``.
```

Reply:
Thank you for the feedback. I've more-or-less taken @HyukjinKwon's suggestion (the comment already mentions that it's equivalent to cast, so I just made that line more obvious/not seem like just a parenthetical comment on the last sentence).